### PR TITLE
bug: Add legacy validate basic logic to govshuttle msgs

### PR DESCRIPTION
**Pull request by dudong2**
## Description

- Added validate basic logic that was missing. (ref. [original validate basic logics](https://github.com/Canto-Network/Canto/blob/972b7c6080fed051140128264e372d610966b621/x/govshuttle/types/proposal.go#L58-L90))

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] included the necessary unit and integration tests
- [ ] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/govshuttle/keeper/msg_server_test.go [hunk 1]
```diff
@@ -86,6 +86,23 @@ func (suite *KeeperTestSuite) TestMsgExecutionByProposal() {
 			func(uint64, sdk.Msg) {},
 			true,
 		},
+		{
+			"fail - MsgLendingMarketProposal - validate basic logic",
```
Comment (zsystm) [inline x/govshuttle/keeper/msg_server_test.go:92]: It seems that the reason this test case is failing is not related to validate basic. Could you please check it?

File: x/govshuttle/keeper/msg_server_test.go [hunk 2]
```diff
@@ -157,6 +174,22 @@ func (suite *KeeperTestSuite) TestMsgExecutionByProposal() {
 			func(uint64, sdk.Msg) {},
 			true,
 		},
+		{
+			"fail - MsgTreasuryProposal - validate basic logic",
```
Comment (zsystm) [inline x/govshuttle/keeper/msg_server_test.go:179]: It seems that the reason this test case is failing is not related to validate basic. Could you please check it?

**Review by zsystm — COMMENTED** [review]
Minor comments for TCs.
LGTM, except for them.

File: x/govshuttle/keeper/msg_server_test.go (hunk 1 above)
Comment (dudong2) [reply x/govshuttle/keeper/msg_server_test.go:92]: I fixed test logic

File: x/govshuttle/keeper/msg_server_test.go (hunk 2 above)
Comment (dudong2) [reply x/govshuttle/keeper/msg_server_test.go:179]: ditto

**Review by zsystm — APPROVED** [review]
LGTM

**Review by poorphd — APPROVED** [review]
LGTM

**Pull request merged**